src/views/movieDetailsView/movieDetailsView.js

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import CustomButton from '../../components/common/customButton/customButton'
+import WikiBadge from '../../components/common/customBadges/customBadges'
+
+import { fetchMovieDetails } from '../../actions/movieDetails';
+import { goBack } from '../../actions/navigation';
+
+import { API_PARAMS } from '../../common/constants'
+
+import './movieDetailsView.scss';
+
+
+class MoviesDetailsView extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showPlot: false
+        };
+    }
+
+    getMovieDetails() {
+        const urlParamId = this.props.match.params.id;
+        const { movieDetails } = this.props.movieDetails;
+
+        if (!movieDetails
+            || (movieDetails.imdbID !== urlParamId)) {
+            const options = {
+                [API_PARAMS.ID]: urlParamId
+            }
+            this.props.fetchMovieDetails(options);
+        }
+    }
+
+    togglePlot = () => {
+        this.setState( (prevState) => {
+            return {
+                showPlot: !prevState.showPlot
+            }
+        });
+    }
+
+    renderPlot = (movieDetails) => {
+        return (
+            <div>
+                <h6><strong>Plot:</strong></h6>
+                <p>
+                    { this.state.showPlot ? movieDetails.Plot : movieDetails.Plot.substr(0, 20) }
+                </p>
+                <p className="btn btn-primary d-inline-block" onClick={this.togglePlot}>
+                    {this.state.showPlot ? 'Hide plot' : 'Show Plot'}
+                </p>
+            </div>
+        );
+    }
+
+    renderFromArray = (data, type = null) => {
+        if (!Array.isArray(data)) {
+            return null;
+        }
+
+        return data.map(item => {
+            if (type === 'list') {
+                return (                                                            
+                    <li key={ `${item.Source}_key` } >
+                        {item.Source}: {item.Value} 
+                    </li>                                                           
+                );
+            }
+
+            return (
+                <WikiBadge
+                    options={{
+                        label: item,
+                        query: (type === 'genre') ? `${item} film` : null
+                    }}
+                    key={ `${item}_key` }/>
+            );
+        })
+    }
+
+    componentWillMount() {
+        this.getMovieDetails();
+    }
+
+    render() {
```

Comment:
What do you think about splitting sections into separate render methods that will be just used in `render()`.
For example renderHeaderInfo for `movieDetailsView__header-info` block,
renderMovieDescription for `movieDetailsView__description` block etc.